src/server/dflycmd.cc

```diff
@@ -395,24 +395,31 @@ void DflyCmd::TakeOver(CmdArgList args, ConnectionContext* cntx) {
 
   VLOG(1) << "AwaitCurrentDispatches done";
 
-  absl::Cleanup([] {
+  absl::Cleanup cleanup([] {
+    VLOG(2) << "Enabling expiration";
     shard_set->RunBriefInParallel([](EngineShard* shard) {
       namespaces.GetDefaultNamespace().GetDbSlice(shard->shard_id()).SetExpireAllowed(true);
     });
-    VLOG(2) << "Enable expiration";
   });
 
+  atomic_bool catchup_success = true;
   if (*status == OpStatus::OK) {
-    auto cb = [replica_ptr = std::move(replica_ptr), end_time, &status](EngineShard* shard) {
-      status = WaitReplicaFlowToCatchup(end_time, std::move(replica_ptr), shard);
```

Comment:
This is the invalid `std::move()` (the one inside the lambda, not the one in the lambda's capture list)

Reply:
can you please add a comment about this (i..e about cb running multiple times)

Reply:
The problem is not _running_ multiple times, the problem is `move()`ing multiple times.
I'll add a comment, sure.

Reply:
maybe the best option will be using simple pointer instead of shared for WaitReplicaFlowToCatchup

Reply:
That's a very good idea. It both the code is better (less complicated, more obvious), and also saves the need for the comment.